From 7554ff2ba43b044ec82f31458c0814ca1cf69bd0 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Mon, 10 Sep 2012 22:12:17 +0200
Subject: [PATCH 30/68] Don't complete command on USB error.

There is no way tie the urb to the command so just let the command
time out instead if there is one.
---
 ar5523.c |    9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -248,25 +248,18 @@ static void ar5523_cmd_rx_cb(struct urb
 	/* sync/async unlink faults aren't errors */
 	if (urb->status && (urb->status != -ENOENT &&
 	    urb->status != -ECONNRESET && urb->status != -ESHUTDOWN)) {
-		ar5523_dbg(ar,
-			   "nonzero write bulk status received: %d\n",
+		ar5523_dbg(ar, "nonzero write bulk status received: %d\n",
 			   urb->status);
-		cmd->res = urb->status;
-		complete(&cmd->done);
 		return;
 	}
 
 	if (urb->status) {
 		ar5523_err(ar, "RX USB error %d.\n", urb->status);
-		cmd->res = urb->status;
-		complete(&cmd->done);
 		return;
 	}
 
 	if (urb->actual_length < sizeof(struct ar5523_cmd_hdr)) {
 		ar5523_err(ar, "RX USB to short.\n");
-		cmd->res = -1;
-		complete(&cmd->done);
 		return;
 	}
 
